### ISSUE TITLE
Attributes target_temp_high and target_temp_low are not always set

### ISSUE THREAD

**Issue by cyberkryten** [swingerman/ha-dual-smart-thermostat#324]
I've been trying to figure out why my temperature updates of the target thermostats keep failing and have finally figured out that the attributes I am reading the set Dual Smart temperature from are vanishing at times.

I am using switches to do this which setup the Underfloor heating thermostat for heating modes and AC system for cooling modes.  Here's the code in use for the Office switches to control the actual devices.:

```
- platform: template
  switches:
    virtual_office_cool:
      value_template: "{{is_state('climate.office_ac', 'cool')}}"
      turn_on:
        - action: climate.set_fan_mode
          data:
            fan_mode: auto
          target:
            entity_id: climate.office_ac
        - action: climate.set_temperature
          data:
            hvac_mode: cool
            temperature: "{{ (state_attr('climate.office_smart', 'target_temp_high') - 3) | round(0, 'floor') }}"
          target:
            entity_id: climate.office_ac
        - action: climate.set_fan_mode
          data:
            fan_mode: high
          target:
            entity_id: climate.office_ac
      turn_off:
        - action: climate.turn_off
          target:
            entity_id: climate.office_ac

    virtual_office_heat:
      value_template: "{{is_state('climate.office', 'heat')}}"
      turn_on:
        - action: climate.set_preset_mode
          target:
            entity_id: climate.office
          data:
            preset_mode: High
        - action: climate.set_temperature
          data:
            temperature: "{{ (state_attr('climate.office_smart', 'target_temp_low') + 3) | round(0, 'ceiling') }}"
          target:
            entity_id: climate.office

      turn_off:
        - action: climate.set_preset_mode
          target:
            entity_id: climate.office
          data:
            preset_mode: Low
        - action: climate.set_temperature
          data:
            temperature: "{{ (state_attr('climate.office_smart', 'target_temp_low') - 5) | round(0, 'floor') }}"
          target:
            entity_id: climate.office
```

Usually, I can read the set values from the documented attributes which are these:

```
target_temp_high: 26
target_temp_low: 22
```

However, at times, these two attributes stop getting set and seem to be replaced with versions prefixed with 'prev_' 

Here's the current state of attributes for my climate.office_smart thermostat, taken from the Developer Tools States tab:

```
hvac_modes: heat_cool, heat, cool, off
min_temp: 12
max_temp: 30
target_temp_step: 1
preset_modes: away, none
current_temperature: 24
temperature: 26
hvac_action: idle
preset_mode: none
prev_target_temp_low: 22
prev_target_temp_high: 28
hvac_action_reason: 
friendly_name: Office Smart
supported_features: 401
```

As you can see, the documented attributes are missing and have been replaced by versions with prev_ on the front - this causes exceptions within the actions because the state_attr call fails to return a value for the current high/low target temperature.

These prev_ versions are not documented.

Sometimes, both are present:

```
hvac_modes: heat_cool, heat, cool, off
min_temp: 12
max_temp: 30
target_temp_step: 1
preset_modes: away, none
current_temperature: 24
target_temp_high: 28
target_temp_low: 26
hvac_action: heating
preset_mode: none
prev_target_temp_low: 26
prev_target_temp_high: 28
hvac_action_reason: 
friendly_name: Office Smart
supported_features: 402
```

**Comment by swingerman** [comment]
Those attributes are only used if range mode is active in the thermostat. In dual mode. If you set your thermostat to heat or cool-only mode, the underlying climate entity does not provide those attributes. We use the "prev_" state attributes not to lose the settings and to set them as when range mode (heat/cool mode) is enabled again.
I would suggest improving your template with `IF` conditions.

**Comment by cyberkryten** [comment]
OK, I can recode the templates to check the state first and then read the appropriate attribute.  I totally missed that the target value being read changes based on the mode - it isn't entirely clear from the Docs that is what happens so I think it could do with some tidying up to avoid confusion.

"Temperature" isn't listed in the Documentation at all in the "Configuration Parameters" section which is why I got confused, so I think it is probably worth updating that to include the three sensors and explaining when each is used.

I think it would also help if it was renamed 'target_temp' to match with the other parameters which all have the target_ prefix on them. That way it would be clearer what it is being used for (though could break tons of stuff!).  


The prev_ storing doesn't actually work.  When changing from AC to heat, the 'target_temp_low' value is used to set 'temperature'.  Then if I increase that, when I go back to AC mode it doesn't re-instate the prev_target_temp_low value, but copies in the 'temperature' value as the new target_temp_low.

Settings in Heat/Cool mode:

```
target_temp_high: 26
target_temp_low: 20

```

Change to Heat only (starts at 20) and change the target to be 23:

```
prev_target_temp_low: 20
prev_target_temp_high: 26
temperature: 23
```

All I did after that was change it to Heat/Cool mode, this is what it ends up with:

```
prev_target_temp_low: 23
prev_target_temp_high: 26
target_temp_high: 26
target_temp_low: 23
```

It didn't restore the '20' as the low bound, but took the 23 from the Heat mode's setting and it does the same when changing to Cool mode and back (but takes/sets the 'high' value instead of the low one).
